scripts/verify-python.sh

```diff
@@ -8,13 +8,19 @@
 #
 # $ ./verify-python.sh --disable=<msg_ids>
 
-run_pylint () {
+list_python_files () {
 	git ls-files \
 		| xargs file \
 		| grep "Python script" \
-		| cut -d ':' -f 1 \
-		| xargs -L 1000 pylint "$@"
+		| cut -d ':' -f 1
 }
 
-pylint --version
-run_pylint --rcfile="$(git rev-parse --show-toplevel)/.pylint" "$@"
+main () {
+	pylint --version >&2
+	echo "Checking files:" >&2
+	list_python_files >&2
+	local -r rc="$(git rev-parse --show-toplevel)/.pylint"
+	list_python_files | xargs -L 1000 pylint --rcfile="$rc" "$@"
```

Comment:
Pylint's output doesn't include the name of the script being checked, so there's no way to connect it's warnings to a specific script:
```
************* Module count-bugs
C: 64, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C: 66, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
************* Module count-warnings
W: 32, 0: Cannot decode using encoding "ascii", unexpected byte at position 32 (invalid-encoded-data)
C: 61, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C: 86, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
C: 88, 0: Unnecessary parens after 'print' keyword (superfluous-parens)
```

Replacing the `-L 1000` with `--verbose -n1` in xargs makes it print the line that's being executed and limits execution to one instance per input file.  So that's one way; but the output looks pretty rough with all the preamble about config files and such (I haven't seen a `quiet` option on pylint either). 

Also, Pylint it a bit heavy to startup.. so I see the advantage of passing everything in at once. Maybe it's config/filters have an option to print the name of the script under analysis?

```
Your code has been rated at 9.37/10 (previous run: 10.00/10, -0.63)
```
The **final score** is only reported from the last file checked (msys-bash.py), as opposed to the prior scripts.

Reply:
I'm not sure what happened here - pylint shipped by fedora, and pylint shipped via pip (I am use pipenv to keep pip installations separate per-project) both give this output:

```
************* Module msys-bash
scripts/msys-bash.py:30:0: C0301: Line too long (140/100) (line-too-long)
scripts/msys-bash.py:1:0: C0103: Module name "msys-bash" doesn't conform to snake_case naming style (invalid-name)
scripts/msys-bash.py:38:4: C0103: Constant name "script_py" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/msys-bash.py:39:4: C0103: Constant name "script_sh" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/msys-bash.py:41:4: C0103: Constant name "bash" doesn't conform to UPPER_CASE naming style (invalid-name)
scripts/msys-bash.py:42:4: C0103: Constant name "rcode" doesn't conform to UPPER_CASE naming style (invalid-name)
```
But documentation for pylint lists this as "older" format (?).

Anyways, the good news is: this is totally configurable via `.pylint` file :) I'll experiment a little, maybe I can make it resemble shellcheck output.

If we'll end up using one pylint invocation per file, then I'll drop scores. Yeah - they are usefult when dealing with projects fixing their styling, but score is meaningless when score 10/10 is enforced from the start.

Reply:
OK, with python3 version of pylint being enforced, better output should be now default.